src/Microsoft.AspNet.Razor/TagHelpers/TagHelperDescriptor.cs

```diff
@@ -147,5 +151,11 @@ public class TagHelperDescriptor
         /// The <see cref="TagHelperUsageDescriptor"/> that indicates how to use the tag helper at design time.
         /// </summary>
         public TagHelperUsageDescriptor UsageDescriptor { get; }
+
+        /// <summary>
+        /// The HTML element the tag helper may output.
+        /// </summary>
+        /// <remarks>Overrides the HTML IntelliSense provided at design time.</remarks>
+        public string OutputElementHint { get; set; }
```

Comment:
`UsageDescriptor`s I feel like should be handled at `DesignTime`. No need to special case them for our purposes. Lot of options there for how to showcase that info to a user.